### PR TITLE
[scatter] few small fixes

### PR DESCRIPTION
**Pull request by chejennifer**
- update stat var hierarchy so that if there's no available stat vars, we still get the header [(issue #1297)](https://github.com/datacommonsorg/website/issues/1297)
    - original: Screen Shot 2021-11-04 at 11 49 49 AM
    - with fix: Screen Shot 2021-11-04 at 11 50 17 AM

- fix [issue #1305](https://github.com/datacommonsorg/website/issues/1305) by combining the useEffect for geoJsonFetched, props, and ChartContainer again

### CODE REVIEW

File: static/js/tools/scatter/chart.tsx [hunk 1]
```diff
@@ -165,26 +165,16 @@ function Chart(props: ChartPropsType): JSX.Element {
...
-  // Replot when chart width changes on sv widget toggle.
+  // Replot when props or chart width changes.
   useEffect(() => {
     const debouncedHandler = _.debounce(() => {
-      if (
-        props.display.chartType === ScatterChartType.SCATTER ||
-        geoJsonFetched
-      ) {
-        replot();
+      if (props.display.chartType === ScatterChartType.MAP && !geoJsonFetched) {
```
Comment (beets) [inline static/js/tools/scatter/chart.tsx:171]: this would also debounce updates on props/geojsonfetched. could you add todo to figure this out?

**Review by beets — APPROVED** [review]

File: static/js/tools/scatter/chart.tsx (hunk 1 above)
Comment (chejennifer) [reply static/js/tools/scatter/chart.tsx]: found a way to only call the debounced method on window resize, updated!

File: static/js/tools/scatter/chart.tsx [hunk 2]
```diff
@@ -165,25 +165,22 @@ function Chart(props: ChartPropsType): JSX.Element {
     }
   }
 
-  // Replot when data changes.
+  // Replot when props or chart width changes.
   useEffect(() => {
+    const entrySet = new Set();
```
Comment (beets) [inline static/js/tools/scatter/chart.tsx:170]: do we need to clear the entryset?

File: static/js/tools/scatter/chart.tsx [hunk 3]
```diff
@@ -165,25 +165,22 @@ function Chart(props: ChartPropsType): JSX.Element {
...
-  }, [props, geoJsonFetched]);
-
-  // Replot when chart width changes on sv widget toggle.
-  useEffect(() => {
-    const debouncedHandler = _.debounce(() => {
-      if (
-        props.display.chartType === ScatterChartType.SCATTER ||
-        geoJsonFetched
-      ) {
+    const debouncedHandler = _.debounce((entries) => {
```
Comment (beets) [inline static/js/tools/scatter/chart.tsx:178]: could you add documentation for entries? i could not find reference to it :-/

File: static/js/tools/scatter/chart.tsx (hunk 1 above)
Comment (beets) [reply static/js/tools/scatter/chart.tsx]: v cool!

File: static/js/tools/scatter/chart.tsx (hunk 3 above)
Comment (chejennifer) [reply static/js/tools/scatter/chart.tsx:180]: done

**Review by beets — APPROVED** [review]

**Pull request merged**